go-controller/pkg/cni/cni.go

```diff
@@ -329,3 +347,38 @@ func (pr *PodRequest) getCNIResult(getter PodInfoGetter, podInterfaceInfo *PodIn
 		IPs:        ips,
 	}, nil
 }
+
+func (pr *PodRequest) newUserDefinedPrimaryNetworkPodRequest(
+	pod *kapi.Pod,
+	userDefinedPrimaryNetwork *udn.UserDefinedPrimaryNetwork,
+) *PodRequest {
+	req := &PodRequest{
```

Comment:
can we fix this TODO: // TODO: if we're actually taking this route, maybe what we want to to build a REQ from the default network REQ since half of it can be copied (pod UID / netns path / ...)
https://github.com/ovn-org/ovn-kubernetes/pull/4459/commits/b2ad9069f4914f3ca2d4b69df0114dcb7a2b77ab#diff-7664be4fc9831cf07ad96809560316efd267b946033659d4e0973b8ceaa674cfR401

Reply:
This is handleing the todo, since it's using the original `pr` to construct the new on
```golang
Command:      pr.Command,
SandboxID:    pr.SandboxID,
Netns:        pr.Netns,
IsVFIO:       pr.IsVFIO,
```